contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -1124,6 +1131,93 @@ validate_special_function(char *func_nsname, char *func_name, List* fargs, int n
 				}
 			}
 		}
+		else if (strlen(func_name) == 6 && strncmp(func_name, "concat", 6) == 0)
+		{
+			Oid		target_type_id = get_sys_varcharoid();
+			/* PG has limitation for max number of args = 100. */
+			if (nargs < 2 || nargs > 100)
+				ereport(ERROR,
+						(errcode(ERRCODE_UNDEFINED_FUNCTION),
+							errmsg("The concat function requires 2 to 100 arguments.")));
+			
+			for (int i = 0; i < nargs; i++)
```

Comment:
can we have arguments with different data types? If so then please point out test cases and if not so then we can break this loop

Reply:
Yes, we can have arguments with different data types. We have various test case added for the same. Please refer to prepare and verify files for the same. One such example: https://github.com/amazon-aurora/babelfish_extensions/blob/4ad48c68062d72aa58bb98e1f12e6961f382e67d/test/JDBC/expected/concat-vu-prepare.out#L472